src/Mods.jl

```diff
@@ -158,17 +160,16 @@ julia> 92%14
 julia> CRT(BigInt, Mod{9223372036854775783}(9223372036854775782), Mod{9223372036854775643}(9223372036854775642))
 ```
 """
-function CRT(remainders, primes)
+function CRT(remainders, primes) where T
     length(remainders) == length(primes) || error("size mismatch")
-    isempty(remainders) && return 1
+    isempty(remainders) && throw(ArgumentError("input arguments should not be empty."))
```

Comment:
A sensible return value would be `zero(eltype(primes))`.

Reply:
It is not intuitive to me whether it should return zero or one for empty inputs. Do you have any arguments favoring zero?

Reply:
BTW, there is another reason why I choose to throw an error. The `eltype` for ntuple is ill-defined, while this interfaces aim to support both.

Reply:
If you specify the result type as the first argument (which defaults to `BigInt`) as you suppose, the type of the return value is clear.
Arguments for `0` as result for empty inputs are for example:

1. the condition `0 <= crt(r, p) < prod(p)` for empty `p` and `prod([]) == 1`
2. for integer arrays, the rest of the algorithm executes and delivers `0`.
3. it is a corner case in the identity `crt([r1; r2], [p1;p2]) == crt(crt(r1, p1), crt(r2, p2)], [lcm(p1); lcm(p2)])`

Reply:
Thanks for detailed explaination, it makes sense. Return 0 instead in the new commit.